### PR TITLE
Frontend Rework - Switch from serde_cyclonedx to cyclonedx-bom

### PR DESCRIPTION
**Pull request by ErinvanderVeen**
# Frontend Rework - Switch from serde_cyclonedx to cyclonedx-bom

Issue: #40 

NOTE: https://github.com/CycloneDX/cyclonedx-rust-cargo/pull/584 starts support for v1.5.
Once upstream fully implements 1.5 we need to re-implement it in our crate.

For now, we merge this PR looking forward to the future.

## Description
This PR completely reworks Genealogos' frontend (called `bom` internally). Similar to the `Backend` and `BackendHandle`, there is now a `Bom` trait that is responsible for the generation of the bom.

## Checklist
Checklist before merging:
- [x] CHANGELOG.md updated
- [x] README.md up-to-date

### CODE REVIEW

**Review by GuillaumeDesforges — COMMENTED** [review]
All these changes look very reasonable to me. Looking forward to the merge :+1:

File: genealogos/src/bom/cyclonedx.rs [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+
+        Ok(bom)
+    }
+}
+
+impl TryFrom<ModelComponent> for Component {
+    type Error = crate::Error;
+
+    fn try_from(model: ModelComponent) -> Result<Self> {
+        Ok(Component {
```
Comment (yannham) [inline genealogos/src/bom/cyclonedx.rs:200]: I don't know if it's feasible, but if `Component` can derive `Default` for free, you might elide a lot of those `foo: None` parameters for free, with a `..Default::default()`. On the other hand, you might have done this on purpose to make sure everything is listed and initialized to what it's supposed to, in which case you can ignore this comment.

**Review by yannham — APPROVED** [review]
Given the size of the diff, I can't pretend I looked at everything with in each and every detail, but from what I can say this looks good to me

File: README.md [hunk 2]
```diff
@@ -43,13 +43,14 @@ For example, using curl, the api can be invoked like this:
 curl "http://localhost:8000/api/analyze?flake_ref=nixpkgs&attribute_path=hello"
 ```
 
-Additionally an optional `cyclonedx_version` query parameter can be provided to specify the CycloneDX version to use.
+Additionally an optional `bom_format` query parameter can be provided to specify the sbom format to use.
```
[the remark this answers is not in the document]
Comment (ErinvanderVeen) [reply README.md:46]: This is a tricky one.

Genealogos can only generate SBOMs (Software BOMs), but CycloneDX is a generic BOM format. You can use CycloneDX for:
- CBOM (Cryptography)
- HBOM (Hardware)
- OBOM (Operation)
- SBOM (Software)
- SaaSBOM (...)
- VDR (Vulnerability Disclosure Report)
- VEX (Vulnerability Exploitability Exchange)

So, in cases where we are talking about the BOM output by Genealogos we refer to it as an SBOM, but when referring to CycloneDX we use "BOM".

We could standardize our usage (probably to SBOM), if you think it that is still preferred knowing the above.

File: genealogos-api/src/main.rs [hunk 3]
```diff
@@ -133,19 +150,26 @@ mod tests {
...
-
-                // 1.5
-                let mut expected_path_1_5 = input_path.clone();
-                expected_path_1_5.set_extension("1_5.out");
-                let expected_output_1_5 = std::fs::read_to_string(expected_path_1_5).unwrap();
+                let response_sbom = match response_json.get("sbom").unwrap() {
+                    serde_json::Value::String(response_sbom) => response_sbom,
+                    _ => panic!("Not a string"),
+                };
+                let response_sbom: serde_json::Value = serde_json::from_str(response_sbom).unwrap();
```
[the remark this answers is not in the document]
Comment (ErinvanderVeen) [reply genealogos-api/src/main.rs:157]: These unwraps are absolutely not okay, but a failure would suggest an error in our tests. Hence the liberal use of them. I would not use unwraps in actual code.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (GuillaumeDesforges) [reply README.md:46]: Let's hope that in the future Nix is pervasive enough that we can generate other things than SBOM :smile: 
`bom` lgtm

File: genealogos/src/bom/cyclonedx.rs [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+use cyclonedx_bom::models::external_reference::ExternalReference;
+use cyclonedx_bom::models::external_reference::ExternalReferenceType;
+use cyclonedx_bom::models::external_reference::ExternalReferences;
+use cyclonedx_bom::models::license::License;
+use cyclonedx_bom::models::license::LicenseChoice;
+use cyclonedx_bom::models::license::LicenseIdentifier;
+use cyclonedx_bom::models::license::Licenses;
+use cyclonedx_bom::models::property::Properties;
+use cyclonedx_bom::models::property::Property;
+use cyclonedx_bom::prelude::*;
```
[the remark this answers is not in the document]
Comment (GuillaumeDesforges) [reply genealogos/src/bom/cyclonedx.rs:16]: Getting PTSD from Java? :stuck_out_tongue:

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ErinvanderVeen) [reply README.md]: I've renamed everything to BOM, including internally. With the exception of a small section in the README

**Pull request merged**